mainpage/wlansi/static/wlansi/js/map.js

```diff
@@ -23,24 +26,44 @@ $(document).ready(function () {
         'mapTypeId': google.maps.MapTypeId.ROADMAP,
     };
     var map = new google.maps.Map(document.getElementById('map_canvas'), map_options);
-    $.each(nodes, function (i, node) {
-        var marker = new google.maps.Marker({
-            'position': new google.maps.LatLng(node.lat, node.long),
+
+    var nodes = [];
+
+    $.get(nodewatcher_url + "/api/v1/stream/?format=json&tags__module=topology&limit=1", function (data) {
+       var streamId = data.objects[0].id;
+       var latestTimestamp = Math.round(Date.parse(data.objects[0].latest_datapoint) / 1000);
+       $.getJSON(nodewatcher_url + "/api/v1/stream/" + streamId + "/?format=json&reverse=true&limit=1&start=" + latestTimestamp, function(x) {
+        nodes = x.datapoints[0].v.v;
+        displayNodes();
+      });
+    });
+
+    function displayNodes() {
+      var n = nodes.length;
+      for (var i = 0; i < n; i++) {
+        node = nodes[i];
+        if (node.hasOwnProperty("l")) {
+          var marker = new google.maps.Marker({
+            'position': new google.maps.LatLng(node.l[1], node.l[0]),
             'map': map,
             'shadow': shadow,
             'icon': icon,
             'shape': shape,
-            'title': node.name
-        });
-        google.maps.event.addListener(marker, 'click', function () {
-            document.location = node.url;
-        });
-    });
+            'title': node.n
+          });
+          (function(uuid){google.maps.event.addListener(marker, 'click', function () {
```

Comment:
Keep coding style. Spaces around:

```
function (uuid) {google.maps.event.addListener(marker, 'click', function () {
```

Also, no need to make one-liners. Target on code readability.

In this particular case you could use `$.each` to iterate over `nodes` which would then solve the problem of variable aliasing in JavaScript.